### ISSUE TITLE
Contribution and credit to upstream project?

### ISSUE THREAD

**Issue by pbuyle** [continuousphp/phing-drush-task#5]
Hi,

This project is a fork of https://www.drupal.org/project/phingdrushtask and fails to acknowledge it. The polite thing to do would be to, at least, mention it and list the original author in the composer.json file.

The change made here are great and more than welcome upstream, sadly, to my knowledge, pushing them to upstream wasn't attempted. This effectively means enhancement made here are lost to existing users of the upstream project, and more generally to most in the Drupal community.

**Comment by fdewinne** [comment]
Hi @pbuyle,

Sorry to not have contact you before... I missed the point :(
Then, please accept my apologies.

The first goal of this fork was to make the project available on composer and packagist. But by the time, some contributions have been included.

I'll try to push them upstream shortly. If you see other change we should add in this fork, don't hesitate!

**Comment by pfrenssen** [comment]
In the meanwhile there is a solution for importing Drupal modules with Composer, it can be done by adding Drupal's own instance of packagist to `composer.json`:

```
    "repositories": [
        {
            "type": "composer",
            "url": "https://packagist.drupal-composer.org"
        },
    ]
```

This has the drawback that composer will fetch those packages too when doing an update, which makes it slower. Alternatively, you can query the repository directly:

```
    "repositories": [
        {
            "type": "package",
            "package": {
                "name": "drupal/phing-drush-task",
                "version": "1.2",
                "source": {
                    "url": "http://git.drupal.org/project/phingdrushtask.git",
                    "type": "git",
                    "reference": "7.x-1.2"
                },
                "autoload": { "classmap": [ "DrushTask.php" ] },
                "include-path": [ "." ]
            }
        }
    ]
```

We could have a look at what has been changed here and propose any necessary changes as contributions upstream, then this fork probably doesn't have a reason to exist any more.

There also have been some important improvements in the original project in the meanwhile. For my use it seems to be perfectly compatible. We have reverted to using the original project on drupal.org in April last year and had absolutely no problems.

**Comment by pbuyle** [comment]
I merged the change from this repo to the 7.x-2.x branch for https://www.drupal.org/project/phingdrushtask

I also have the impression that the changes made here are still perfectly compatible for most users.

**Comment by fdewinne** [comment]
Yes I think so. And more interesting for @pfrenssen is that from now you could remove the autoload part. It's now handled by the package itself

**Comment by pbuyle** [comment]
The dev version of 7.x-2.x is now available at https://packagist.drupal-composer.org/packages/drupal/phingdrushtask

**Comment by pbuyle** [comment]
@fdewinne I made you maintainer on the drupal.org, I'm not maintaining the task for the moment has we have swtiched to a non-phing workflow since a few months now. I also pushed the 7.x-2.x branch on Github.

If you wish to continue using Guthub, I suggest updating the project on drupal.org so it invite people to contribute on GH. Then the next step is likely to do remove the 2.0 release here and on drupal.org to have everything in sync. Once this is done, it is only a matter of keeping the dev. branch in sync between the two repo and using the right tag when releasing (on drupal.org, you still have to actually crate a release from the tag)
